runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -481,14 +502,24 @@ def ssh(self, node: str = None):
             >>> rh.ondemand_cluster("rh-cpu").ssh()
             >>> rh.ondemand_cluster("rh-cpu", node="3.89.174.234").ssh()
         """
-        if node is None:
-            # SSH onto head node - can provide the name as specified in the local ~/.ssh/config
-            subprocess.run(["ssh", f"{self.name}"])
-        else:
-            # If SSHing onto a specific node, which requires the default sky public key for verification
-            sky_key = Path(self.DEFAULT_KEYFILE).expanduser()
 
-            if not sky_key.exists():
-                raise FileNotFoundError(f"Expected default sky key in path: {sky_key}")
+        if self.provider == "kubernetes":
+            command = f"kubectl get pods | grep {self.name}"
+
+            try:
+
+                output = subprocess.check_output(command, shell=True, text=True)
 
-            subprocess.run(["ssh", "-i", str(sky_key), f"ubuntu@{node}"])
```

Comment:
+1, we dropped node support here. I would leave the existing logic alone for now in the case where `self.provider != "kubernetes"`, and we should update it in a separate PR anyway because skypilot now has nicer support for sshing into nodes (you can just do `ssh my-cluster-worker1` which is pretty neat, so we could accept an int for node in addition to an ip).